components/availability/src/Availability.svelte

```diff
@@ -646,6 +723,189 @@
     }
   }
   // #endregion Date Change
+
+  //#region dragging
+
+  // Click + Drag has two basic modes:
+  // - when you click + drag on an available spot, it should create a new series of selected timeSlots
+  // - when YOu click + drag on an already-selected spot, if should let you move that spot
+  // A non-dragging click will either select the timeslot you clicked, or if it's already a selected block, delete that block.
+  // Note some fun behaviour: If you drag a long event overtop a busy spot, it will automatically split it into multiple events.
+  // You can set partial_bookable_ratio to 0 in order to prevent this / let you book atop busy spots.
+  let mouseIsDown = false;
+  let dragStartSlot: SelectableSlot | null = null;
+  let dragStartDay: Day | null = null;
+  let draggingExistingBlock: boolean = false;
+
+  function resetDragState() {
+    days.forEach((day) =>
+      day.slots
+        .filter((x) => x.selectionPending)
+        .forEach((x) => {
+          x.selectionPending = false;
+        }),
+    );
```

Comment:
since this exact code is used elsewhere, perhaps it can be made into a function?